src/components/Accordion/Accordion.test.tsx

```diff
@@ -103,4 +103,31 @@ describe('<Accordion />', () => {
     expect(onOpen).toHaveBeenCalledTimes(1);
     expect(onClose).not.toHaveBeenCalled();
   });
+
+  it('should not call onOpen callback when accordion opens on an empty row', async () => {
+    const user = userEvent.setup();
+    const onClose = jest.fn();
+    const onOpen = jest.fn();
+    render(
+      <Accordion headingAs="h2">
+        <Accordion.Row isExpandable={false}>
+          <Accordion.Button
+            data-testid="accordion-button"
+            onClose={onClose}
+            onOpen={onOpen}
+          >
+            Accordion Button
+          </Accordion.Button>
+          <Accordion.Panel>Accordion Panel</Accordion.Panel>
+        </Accordion.Row>
+      </Accordion>,
+    );
+    const accordionButton = screen.getByTestId('accordion-button');
```

Comment:
Nit: per [testing libraries query priorities](https://testing-library.com/docs/queries/about/#priority), `getByRole` is preferred if it's possible to use.

Reply:
good callout. will see if there's anything blocking use of "role"

Reply:
Ohh, i remember why I deliberately used testId for these ones. at one point, data attributes were not making it to the DOM, and I added some tests to show that it was working. That's mostly a given, so the tests were only so useful in the long term